webui/react/src/pages/TrialDetails/useTrialMetrics.ts

```diff
@@ -69,13 +70,17 @@ const summarizedMetricToSeries = (
     };
     trialData[metricToKey(metric)] = series;
   });
-  // Check to see if any metric contains at least one value for any
-  // xAxis option.
   const xAxisOptions = Object.values(XAxisDomain);
-  const hasData = Object.keys(trialData).some((key) =>
-    xAxisOptions.some((xAxis) => (trialData?.[key]?.data?.[xAxis]?.length ?? 0) > 0),
-  );
-  return { data: trialData, hasData };
+  // Record whether or not each metric contains at least one value for any
+  // xAxis option.
+  Object.keys(trialData).forEach((key) => {
+    if (!metricHasData?.[key]) {
+      metricHasData[key] = xAxisOptions.some(
+        (xAxis) => (trialData?.[key]?.data?.[xAxis]?.length ?? 0) > 0,
+      );
+    }
```

Comment:
see above re: or operator:
```suggestion
    metricHasData[key] ||= xAxisOptions.some(
      (xAxis) => (trialData?.[key]?.data?.[xAxis]?.length ?? 0) > 0,
    );
```